ShoppingListApi/Controllers/PersonController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PersonPetsApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PersonController : ControllerBase
+    {
+        private static List<Person> _people = new List<Person>();
+
+        [HttpPost]
+        public IActionResult Create(Person person)
+        {
+            _people.Add(person);
+
+            return Created("/person", person);
+        }
+
+        [HttpGet]
+        public IActionResult Get()
+        {
+            return Ok(_people);
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult Get(int id)
+        {
+            var person = FindPerson(id);
+
+            if (person == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+
+            }
+
+            return Ok(person);
+        }
+
+        [HttpDelete("{id}")]
+        public IActionResult DeletePerson(int id)
+        {
+            var person = FindPerson(id);
+
+            if (person == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            _people.Remove(person);
+
+            return Ok();
+        }
+
+        [HttpDelete("{id}/{name}")]
+        public IActionResult DeletePet(int id, string name)
+        {
+            var person = FindPerson(id);
+
+            if (person == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+            else
+            {
+                if (person.Pets.Any(x => x.Name == name))
+                {
+                    person.Pets.Remove(person.Pets.Where(x => x.Name == name).FirstOrDefault());
+
+                    return Ok();
+                }
+            }
+
+            return NotFound($"Pet by name {name} was not found for person {id}.");
+        }
+
+        [HttpPatch("{personId}")]
+        public IActionResult AddPet(int personId, Pet pet)
+        {
+            var person = FindPerson(personId);
+
+            if (person == null)
+            {
+                return NotFound($"Person by id {personId} was not found.");
+            }
+
+            person.Pets.Add(pet);
+
+            return Ok();
+        }
+
+        [HttpPut("{id}")]
+        public IActionResult UpdatePerson(int id, Person person)
+        {
+            var oldPerson = FindPerson(id);
+
+            if (oldPerson == null)
+            {
+                return NotFound($"Person by id {id} was not found.");
+            }
+
+            oldPerson.Pets = person.Pets;
+            oldPerson.Name = person.Name;
+            oldPerson.Age = person.Age;
+            oldPerson.SpouseId = person.SpouseId;
```

Comment:
Out of place. Move this to a person class.

Reply:
Moved this logic to Person class.